### ISSUE TITLE
I think we need to test RamTorch

### ISSUE THREAD

**Issue by FurkanGozukara** [kohya-ss/musubi-tuner#749]
I am using AI Toolkit of Ostris right now 

It has RamTorch maybe you already know @kohya-ss 

Using it on Z Image Ultra model

with 100% offloading the VRAM usage drops 22 GB to 11.2 GB 

The speed drop is from 6.7 second / it to 9 second / it

The speed drop looks like lower than block swapping

Also Ostris AI Toolkit Z Image Ultra training speed on Linux and Windows 100% same, but it might be due to it is underoptimized , it wasnt 100% using GPU on either platform

I think you can run some tests before deciding to implement or not

**Comment by kohya-ss** [comment]
Thank you! I think that is Z-Image Turbo instead of Ultra.

I have already tested RamTorch and confirmed that it is as fast as block swap.

I will try implementing it with block swap, and if there is a slowdown or large VRAM usage with it, I will consider introducing RamTorch.

Please let me know the resolution, batch size, and GPU you used.

**Comment by FurkanGozukara** [comment]
> Thank you! I think that is Z-Image Turbo instead of Ultra.
> 
> I have already tested RamTorch and confirmed that it is as fast as block swap.
> 
> I will try implementing it with block swap, and if there is a slowdown or large VRAM usage with it, I will consider introducing RamTorch.
> 
> Please let me know the resolution, batch size, and GPU you used.

Yes Turbo model my bad

I have done test with this way

all images are 1536x1536
training resolution : 1536 - no other resolutions
batch size 1 
lora network ranks are 128
no FP8 quantization 
text  encoders are deloaded after cached

tested on RTX 5090 on Windows

interestingly my Windows speed is exactly same as Linux. With Musubi Tuner Linux is always significantly faster :D 

used AdamW8Bit

**Comment by kohya-ss** [comment]
I tested it on an RTX 6000 Pro Max-Q in a Windows environment, 1536x1536, batch size 1, rank=128, bf16 mixed precision, gradient checkpointing is enabled, with FlashAttention 2.

No block swap, without torch.compile: VRAM usage 19,551MB, about 4.5sec/it
No block swap, with torch.compile (dynamic=true): VRAM usage 19,551MB, about 4.4sec/it
Full block swap (28), use_pinned_memory_for_block_swap: VRAM usage 10,156MB, about 4.7sec/it

There may be differences in the modules being trained (noise_refiner and context_refiner are not being trained on Musubi Tuner), but VRAM consumption is kept low enough, and the speed seems to be sufficient.

I haven't tested it with the WDDM mode driver, but I think we can expect decent speeds.

**Comment by FurkanGozukara** [comment]
> I tested it on an RTX 6000 Pro Max-Q in a Windows environment, 1536x1536, batch size 1, rank=128, bf16 mixed precision, gradient checkpointing is enabled, with FlashAttention 2.
> 
> No block swap, without torch.compile: VRAM usage 19,551MB, about 4.5sec/it No block swap, with torch.compile (dynamic=true): VRAM usage 19,551MB, about 4.4sec/it Full block swap (28), use_pinned_memory_for_block_swap: VRAM usage 10,156MB, about 4.7sec/it
> 
> There may be differences in the modules being trained (noise_refiner and context_refiner are not being trained on Musubi Tuner), but VRAM consumption is kept low enough, and the speed seems to be sufficient.
> 
> I haven't tested it with the WDDM mode driver, but I think we can expect decent speeds.

thank you so much your hard work is amazing. appreicate that
